cli/waiter/subcommands/ping.py

```diff
@@ -1,87 +1,77 @@
+            read_timeout = timeout_seconds * 2
+            resp = http_util.get(cluster, '/waiter-ping', headers=headers, read_timeout=read_timeout)
             logging.debug(f'Response status code: {resp.status_code}')
+            resp_json = resp.json()
             if resp.status_code == 200:
-                print(terminal.success(f'Ping successful.'))
-                print(resp.text)
-                ping_result = PingResult.SUCCESS
+                status = resp_json['service-state']['status']
+                ping_response = resp_json['ping-response']
+                ping_response_result = ping_response['result']
+                if ping_response_result == 'received-response':
+                    ping_response_status = ping_response['status']
+                    if ping_response_status == 200:
+                        print(terminal.success('Ping successful.'))
+                        result = True
+                    else:
+                        print_error('Ping responded with non-200 status.')
+                        result = False
+                elif ping_response_result == 'timed-out':
+                    if wait_for_request:
+                        print_error('Ping request timed out.')
+                        result = False
+                    else:
+                        logging.debug('ignoring ping request timeout due to --no-wait')
+                        result = True
+                else:
+                    print_error(f'Encountered unknown ping result: {ping_response_result}.')
+                    result = False
             else:
-                print_error(response_message(resp.json()))
-                ping_result = PingResult.FAILURE
+                print_error(response_message(resp_json))
+                result = False
         except Exception:
-            ping_result = PingResult.ERRORED
+            result = False
             message = f'Encountered error while pinging in {cluster_name}.'
             logging.exception(message)
             if wait_for_request:
                 print_error(message)
 
-    def service_exists():
-        logging.debug(f'Ping result: {ping_result}')
-        if ping_result.failed():
-            return {'result': ping_result}
-        else:
-            return service_exists_fn()
+        return result, status
 
     def check_service_status():
-        result = wait_until(service_exists, timeout=timeout, interval=5)
+        result = wait_until(service_exists_fn, timeout=timeout, interval=5)
         if result:
-            if ping_result.failed():
-                return False
-            else:
-                print(f'Service is currently {format_status(result["status"])}.')
-                return True
+            print(f'Service is currently {format_status(result["status"])}.')
+            return True
         else:
             print_error('Timeout waiting for service to exist.')
```

Comment:
What do you think about: `Timeout while trying to start service`?

Reply:
Changed this to `Timeout while waiting for service to start.`, let me know what you think